automation/taskcluster/lib/tasks.py

```diff
@@ -442,6 +454,70 @@ def craft_upload_apk_nimbledroid_task(self, assemble_task_id):
             dependencies=[assemble_task_id],
         )
 
+    def craft_raptor_task(self, signing_task_id, mozharness_task_id, variant):
+        build_location = '{}/{}/artifacts/{}'.format(
+            _DEFAULT_TASK_URL, signing_task_id, DEFAULT_APK_ARTIFACT_LOCATION
+        )
+        architecture, _, __ = get_architecture_and_build_type_and_product_from_variant(variant)
+        worker_type = 'gecko-t-ap-perf-g5' if architecture == 'arm' else 'gecko-t-ap-perf-p2'
+
+        gecko_revision = taskcluster.Queue().task(mozharness_task_id)['payload']['env']['GECKO_HEAD_REV']
+
+        return self._craft_default_task_definition(
+            worker_type=worker_type,
+            provisioner_id='proj-autophone',
+            dependencies=[signing_task_id],
+            name='raptor speedometer: {}'.format(variant),
+            description='Raptor Speedometer on the Reference Browser',
+            payload={
+                "artifacts": [{
+                    'path': '/builds/worker/{}'.format(worker_path),
+                    'expires': taskcluster.stringDate(taskcluster.fromNow(DEFAULT_EXPIRES_IN)),
+                    'type': 'directory',
+                    'name': 'public/{}/'.format(public_folder)
+                } for worker_path, public_folder in (
+                    ('artifacts', 'test'),
+                    ('workspace/build/logs', 'logs'),
+                    ('workspace/build/blobber_upload_dir', 'test_info')
+                )],
+                "command": [
+                    "./test-linux.sh",
+                    '--installer-url={}'.format(build_location),
+                    "--test-packages-url={}/{}/artifacts/public/build/target.test_packages.json".format(_DEFAULT_TASK_URL, mozharness_task_id),
+                    "--test=raptor-speedometer",
+                    "--app=refbrow",
```

Comment:
Can we use the full name here? `reference-browser` or `reference_browser` if dashes can't work?

Reply:
Not at the moment :/ https://searchfox.org/mozilla-central/rev/8d78f219702286c873860f39f9ed78bad1a6d062/taskcluster/taskgraph/transforms/tests.py#839